addons/survey/models/survey_survey.py

```diff
@@ -568,6 +568,32 @@ def _get_number_of_attempts_lefts(self, partner, email, invite_token):
     # QUESTIONS MANAGEMENT
     # ------------------------------------------------------------
 
+    def _remove_invalid_conditional_questions(self, questions):
+        possible_trigger_questions = self.env['survey.question'].search([
+            ('is_page', '=', False),
+            ('question_type', 'in', ['simple_choice', 'multiple_choice']),
+            ('suggested_answer_ids', '!=', False),
+            ('survey_id', '=', self.id)
+        ])
+        valid_questions = self.env['survey.question']
+        invalid_questions = self.env['survey.question']
+        for question in questions:
+            if not question.is_conditional:
+                valid_questions |= question
+                continue
+            if question.triggering_question_id in invalid_questions:
+                invalid_questions |= question
+                continue
+            allowed_triggering_question_ids = possible_trigger_questions.filtered(
+                lambda q: q.sequence < question.sequence or q.sequence == question.sequence and q.id < question.id
+            )
```

Comment:
Some simplifications seem possible for this case with only one survey and existing records, which shouldn't require you to
*  iterate `possible_trigger_questions` inside the question loop. I think a single loop over `questions` should suffice as we want to support conditions going one way and not the other when filling-in the survey.
* build `valid_questions` before the end, you can just return `self - invalid_questions`.